SwiftCompilerSources/Sources/Optimizer/Utilities/OptUtils.swift

```diff
@@ -839,6 +839,7 @@ extension SimplifyContext {
       }
     }
 
+    first.salvageDebugInfo(self)
```

Comment:
I'm wondering why this is needed.
The following `replace` calls `erase(instruction:)` which itself calls salvageDebugInfo

Reply:
Ah, okay